streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1344,19 +1360,30 @@ private <S extends StateStore> InternalTopicConfig createChangelogTopicConfig(fi
     }
 
     public boolean hasOffsetResetOverrides() {
-        return !(earliestResetTopics.isEmpty() && earliestResetPatterns.isEmpty()
-            && latestResetTopics.isEmpty() && latestResetPatterns.isEmpty());
+        return noneResetTopics.size() + noneResetPatterns.size()
+            + earliestResetTopics.size() + earliestResetPatterns.size()
+            + latestResetTopics.size() + latestResetPatterns.size()
+            + durationResetTopics.size() + durationResetPatterns.size() > 0;
     }
 
     public AutoOffsetResetStrategy offsetResetStrategy(final String topic) {
-        if (maybeDecorateInternalSourceTopics(earliestResetTopics).contains(topic) ||
+        final Set<Duration> resetDuration = new HashSet<>();
+
+        if (maybeDecorateInternalSourceTopics(noneResetTopics).contains(topic) ||
+            noneResetPatterns.stream().anyMatch(p -> p.matcher(topic).matches())) {
+            return AutoOffsetResetStrategy.NONE;
+        } else if (maybeDecorateInternalSourceTopics(earliestResetTopics).contains(topic) ||
             earliestResetPatterns.stream().anyMatch(p -> p.matcher(topic).matches())) {
             return AutoOffsetResetStrategy.EARLIEST;
         } else if (maybeDecorateInternalSourceTopics(latestResetTopics).contains(topic) ||
             latestResetPatterns.stream().anyMatch(p -> p.matcher(topic).matches())) {
             return AutoOffsetResetStrategy.LATEST;
+        } else if (maybeDecorateInternalSourceTopics(durationResetTopics.keySet()).contains(topic)) {
+            return AutoOffsetResetStrategy.fromString("by_duration:" + durationResetTopics.get(topic).toString());
+        } else if (findDuration(topic, resetDuration)) {
```

Comment:
I'm thinking this could be simplified (part 1, part 2 in next comment below)

```
 public AutoOffsetResetStrategy offsetResetStrategy(final String topic) {
       final Optional<Duration> duration;
       if (...) {
           ....
       } else if ((duration = findDuration(topic)).isPresent()) {
                 return AutoOffsetResetStrategy.fromString("by_duration:" + duration);
      }
    ...
```